bpf: Only create veth pair if it does not already exist.

jrajahalme opened the pull request:
Deleting and re-creating a veth pair will pick up new MAC addresses
and interface indices, avoid that by figuring out if the veth pair
already exists.

This fixes a frequest CI issue, but also hides underlying problems
relating to endpoint bpf regeneration on daemon config changes or
restarts. New GH issues are opened to keep track of those.

Fixes: #4198
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/4214)
<!-- Reviewable:end -->

rlenglet commented on bpf/init.sh:
```diff
@@ -86,8 +86,12 @@ function setup_veth_pair()
-	ip link del $NAME1 2> /dev/null || true
-	ip link add $NAME1 type veth peer name $NAME2
+	# Only recreate the veth pair if it does not exist already.
+	# This avoids propblems with changing MAC addresses.
```
propblems -> problems

jrajahalme replied:
Fixed, thanks!

jrajahalme replied:
First CI build succeeded :-)

joestringer reviewed (approved):
We reviewed this offline together to validate that `set -e` won't cause weird behaviour inside the if statement. Looks correct, a green CI build should provide reasonable assurance that it doesn't cause regressions.

jrajahalme committed:
bpf: Only create veth pair if it does not already exist.

Deleting and re-creating a veth pair will pick up new MAC addresses
and interface indices, avoid that by figuring out if the veth pair
already exists.

This fixes a frequest CI issue, but also hides underlying problems
relating to endpoint bpf regeneration on daemon config changes or
restarts. New GH issues are opened to keep track of those.

Fixes: #4198
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
